crates/polars-io/src/ndjson/core.rs

```diff
@@ -406,23 +405,25 @@ struct Scratch {
     buffers: simd_json::Buffers,
 }
 
+fn json_lines(bytes: &[u8]) -> impl Iterator<Item = &[u8]> {
+    // This previously used `serde_json`'s `RawValue` to deserialize chunks without really deserializing them.
+    // However, this convenience comes at a cost. serde_json allocates and parses and does UTF-8 validation, all
+    // things we don't need since we use simd_json for them. Also, `serde_json::StreamDeserializer` has a more
+    // ambitious goal: it wants to parse potentially *non-delimited* sequences of JSON values, while we know
+    // our values are line-delimited. Turns out, custom splitting is very easy, and gives a very nice performance boost.
+    bytes.split(|&byte| byte == b'\n').filter(|&bytes| {
```

Comment:
Does `NDJSON` not support lines delimiter within string fields? E.g.

`{"My text with a line split.\nAnd this is the second line."}`.


If it does, we could also reuse `split_lines` iterator from the csv logic, which checks that properly.

Reply:
Per [the specification](https://github.com/ndjson/ndjson-spec?tab=readme-ov-file#31-serialization), this is incorrect (newlines in strings needs to be escaped). Of course, that doesn't mean users are not relying on this, which may mean we want to support that too :)

Reply:
Also, the current line split for threads does not check this (it only checks for `}\n`), which means that users relying on this have a subtle bug that may manifest only occasionally.

Reply:
Ok, then leave it like this. I want us to follow specs (or the closest thing there is to a spec, looking at csv).